substrate/frame/assets-freezer/src/impls.rs

```diff
@@ -0,0 +1,165 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::*;
+
+use frame_support::traits::{
+	fungibles::{Inspect, InspectFreeze, MutateFreeze},
+	tokens::{DepositConsequence, Fortitude, Preservation, Provenance, WithdrawConsequence},
+};
+use pallet_assets::FrozenBalance;
+use sp_runtime::traits::Zero;
+
+/// Implements [`FrozenBalance`] from [`pallet-assets`], so it can understands how much of an
+/// account balance is frozen, and is able to signal to this pallet when to clear the state of an
+/// account.
+impl<T: Config<I>, I: 'static> FrozenBalance<AssetIdOf<T, I>, AccountIdOf<T>, AssetBalanceOf<T, I>>
+	for Pallet<T, I>
+{
+	fn frozen_balance(
+		asset: AssetIdOf<T, I>,
+		who: &AccountIdOf<T>,
+	) -> Option<AssetBalanceOf<T, I>> {
+		FrozenBalances::<T, I>::get(asset, who)
+	}
+
+	fn died(asset: AssetIdOf<T, I>, who: &AccountIdOf<T>) {
+		FrozenBalances::<T, I>::remove(asset.clone(), who);
+		Freezes::<T, I>::remove(asset, who);
+	}
+}
+
+/// Implement [`fungibles::Inspect`](frame_support::traits::fungibles::Inspect) as it is bound by
+/// [`fungibles::InspectFreeze`](frame_support::traits::fungibles::InspectFreeze) and
+/// [`fungibles::MutateFreeze`](frame_support::traits::fungibles::MutateFreeze). To do so, we'll
+/// re-export all of `pallet-assets` implementation of the same trait.
```

Comment:
sounds more like a dev comment, not an end user (still dev) doc

Reply:
Right. It's more like a mention for maintainers. If you consider it's unnecessary, I might just deleted it.

Reply:
it can be rephrased a bit to be more general doc like. Or specifically in this words it can be just a comment for devs, not a doc.

Reply:
Good idea! I left it as a comment for devs instead of a doc.